Best practices/Vulnerable dependencies

```diff
@@ -0,0 +1,69 @@
+
+These reports are not always easily understandable and might take a lot of time to grasp. In some cases, the impact might be obvious, but in a lot of them, the vulnerability will exploit a tiny feature in a transitive dependency which might be hard to figure out if it is being used in our solution. Because of that, it might be advisable not to go down the investigation rabbit hole and continue with the next step.
+
+### Fixing the vulnerabilities
+
+The process of fixing these issues is, in theory, simple: just replace all the vulnerable dependencies with newer versions that have fixed the issue, or find a replacement package that doesn't have the vulnerability. In practice, unfortunately, this is not always easy to do. Fnding out which exact package we should update is not as simple as updating the packages used by our projects because we don't just check the packages referenced directly by our projects, but also the transitive dependencies. Currently, there is no native way (integrated into Visual Studio or the `dotnet` CLI tool) to find out which one of our direct dependencies uses the problematic transitive dependency. Fortunately, we weren't the first ones annoyed by this lack of support.
+
+#### Nail Down NuGet
+
+[NailDownNuGet](https://github.com/Kraego/NailDownNuget) is a PowerShell script which uses [`nuget-deps-tree`](https://www.npmjs.com/package/nuget-deps-tree) (available on `npm`) to get the dependency tree and then traverses through it to determine which package uses the transitive dependency we're searching for.
+
+To use the script, we must first install the `nuget-deps-tree` package:
+
+``` powershell
+npm install nuget-deps-tree
+```
+
+After we've installed the package, we can run the script from the repository. To run the unsigned script, you might need to open the PowerShell in Bypass execution policy mode (you can also set the policy permanently, but that is not advisable):
+
+``` powershell
+powershell.exe -executionpolicy Bypass
+```
+
+To run the script, use the following command:
+
+``` powershell
+./nailDownNuget.ps1 <path-to-sln> <package-name> <package-version>
+```
+
+This will print all the dependency paths which include the specified package and version. Using that information we can determine which dependencies include the vulnerable one:
```

Comment:
`This will print all dependency paths ...` instead of `This will print all the dependency paths...` ?